scripts/scil_compute_bundle_profiles.py

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""
+Compute tract profiles and their statistics along streamlines.
+"""
+
+
+import argparse
+import json
+import os
+
+import nibabel as nib
+import numpy as np
+
+
+from scilpy.io.streamlines import load_tractogram_with_reference
+from scilpy.io.image import assert_same_resolution
+from scilpy.io.utils import (assert_inputs_exist,
+                             add_json_args,
+                             add_overwrite_arg,
+                             add_reference_arg)
+from scilpy.utils.filenames import split_name_with_nii
+from scilpy.utils.metrics_tools import get_bundle_metrics_profiles
+from scilpy.tracking.tools import resample_streamlines_num_points
+from scilpy.tractanalysis.features import get_streamlines_centroid
+
+
+def norm_l2(x):
+    return np.sqrt(np.sum(np.power(x, 2), axis=1, dtype=np.float32))
+
+
+def average_euclidean(x, y):
+    return np.mean(norm_l2(x - y))
+
+
+def _build_arg_parser():
+    p = argparse.ArgumentParser(formatter_class=argparse.RawTextHelpFormatter,
+                                description=__doc__)
+    p.add_argument('in_bundle',
+                   help='Fiber bundle file to compute the tract profiles on.')
+    p.add_argument('in_metrics', nargs='+',
+                   help='Nifti metric(s) on which to compute '
+                        'the tract profiles.')
+    p.add_argument('--in_centroid',
+                   help='If provided it will be used to make sure all '
+                        'streamlines go in the same direction. Otherwise, a '
+                        'centroid will be automatically computed.')
```

Comment:
This script breaks if centroid does not have the right number of --nb_pts_per_streamline.

Please robustify.

Also, even if centroid has the right number of points it breaks right now. See data here:
https://www.dropbox.com/s/xrra0c3084xvlyu/bundle_profile_bug.zip?dl=0

 scil_compute_bundle_profiles.py output/AF_L_m.trk --in_centroid AF_L_centroid_10pts.trk  t1_converted.nii.gz --nb_pts_per_streamline 10
scil_compute_bundle_profiles.py output/AF_L_m.trk --in_centroid AF_L_centroid.trk  t1_converted.nii.gz --nb_pts_per_streamline 10

Reply:
Do you want to force resampling of the centroid if `in_centroid and --nb_pts_per_streamline` or raise an error ?

Reply:
I personally think they should be mutually exclusive to be really clear (but required). If nothing is provided we computed it with the nbr_of_points, and if we provide the centroids we use them as is.
Anyway if we want to resample centroids we should use `scil_resample_streamlines.py`

I would also be ok with removing that parameter entirely, since we rarely call that script out of the blue, we usually follow steps which include `scil_compute_centroids.py` (like in the tractometry pipeline)

Reply:
Everything should be fixed.
@max your centroid has 99 points.